### PR TITLE
picasa/slurp: use the redirect-less flow instead

### PR DESCRIPTION
**Pull request by aboodman**

### CODE REVIEW

File: samples/js/picasa/slurp/src/oauth.js [hunk 1]
```diff
@@ -56,35 +54,9 @@ export function getAccessTokenFromRefreshToken(
...
-      } finally {
-        const message = code ? `got code ${code}` : 'failed to get code';
-        response.end(`<body>${message}</body>`);
-      }
-    }).listen(0);
-
+  return new Promise(res => {
     const secret = String(Math.random());
-    const redirectUri = `http://localhost:${server.address().port}`;
+    const redirectUri = 'urn:ietf:wg:oauth:2.0:oob';
```
Comment (kalman) [inline samples/js/picasa/slurp/src/oauth.js]: comment

File: samples/js/picasa/slurp/src/oauth.js [hunk 2]
```diff
@@ -93,7 +65,17 @@ function getAuthCodeViaURL(clientId: string, clientSecret: string)
...
     });
-    console.log(authUrl);
+
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+    rl.question(
+      `Visit the following URL and paste the code you get:\n\n\t${authUrl}\n\n`,
+      (answer) => {
```
Comment (kalman) [inline samples/js/picasa/slurp/src/oauth.js]: you can do `answer` not `(answer)`

File: samples/js/picasa/slurp/src/oauth.js [hunk 3]
```diff
@@ -56,35 +54,9 @@ export function getAccessTokenFromRefreshToken(
 
 function getAuthCodeViaURL(clientId: string, clientSecret: string)
```
Comment (kalman) [inline samples/js/picasa/slurp/src/oauth.js]: function needs rename

File: samples/js/picasa/slurp/src/oauth.js [hunk 4]
```diff
@@ -93,7 +65,17 @@ function getAuthCodeViaURL(clientId: string, clientSecret: string)
...
-    console.log(authUrl);
+
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+    rl.question(
+      `Visit the following URL and paste the code you get:\n\n\t${authUrl}\n\n`,
+      (answer) => {
+        res([answer, redirectUri]);
```
Comment (kalman) [inline samples/js/picasa/slurp/src/oauth.js]: now that the redirect URI is a constant, you could define `'urn:ietf:wg:oauth:2.0:oob'` as a global const, then get rid of the need to pass `redirectUri` through the promise resolve.

File: samples/js/picasa/slurp/src/oauth.js (hunk 3 above)
Comment (aboodman) [inline samples/js/picasa/slurp/src/oauth.js]: done

File: samples/js/picasa/slurp/src/oauth.js (hunk 1 above)
Comment (aboodman) [inline samples/js/picasa/slurp/src/oauth.js]: done

File: samples/js/picasa/slurp/src/oauth.js (hunk 2 above)
Comment (aboodman) [inline samples/js/picasa/slurp/src/oauth.js]: done

**Pull request merged**